### PR TITLE
Add reason for server is busy error

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
@siddontang @BusyJay PTAL

### CODE REVIEW

File: src/raftstore/errors.rs [hunk 1]
```diff
@@ -174,7 +174,9 @@ impl Into<errorpb::Error> for Error {
                 errorpb.set_stale_command(errorpb::StaleCommand::new());
             }
             Error::Transport(transport::Error::Discard(_)) => {
-                errorpb.set_server_is_busy(errorpb::ServerIsBusy::new());
+                let mut server_is_busy_err = errorpb::ServerIsBusy::new();
+                server_is_busy_err.set_reason(String::from("raftstore is busy"));
```
Comment (siddontang) [inline src/raftstore/errors.rs]: I prefer defining the static const &str and use to_owned() here.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**